dot-net-sdk/EppoClient.cs

```diff
@@ -400,6 +402,45 @@ public static EppoClient Init(EppoClientConfig eppoClientConfig)
         return new(variation);
 
     }
+
+    public void LogNonBanditAction(string subjectKey,
+                                   string flagKey,
+                                   Dictionary<string, object?> subjectAttributes,
+                                   string actionString,
+                                   Dictionary<string, object?> actionAttributes)
+    {
+        // Exception loggingException = null;
+
+        // Dev is allowed to pass nulls to `GetBanditAction` but not the flag assignment methods. Something to fix in the future.
+        var subjectAttributesNoNulls = subjectAttributes.Where(kvp => kvp.Value != null).ToDictionary(kvp => kvp.Key, kvp => kvp.Value!);
+        var assignmentResult = GetStringAssignment(flagKey, subjectKey, subjectAttributesNoNulls, AssignmentDefaultValue);
+
+        if (assignmentResult == AssignmentDefaultValue)
+        {
+            // No variation is assigned, neither bandit nor control.
+            return;
+        }
+
+        var subject = ContextAttributes.FromDict(subjectKey, subjectAttributes);
+        var action = ContextAttributes.FromDict(actionString, actionAttributes);
+
+        _eppoClientConfig.AssignmentLogger.LogBanditAction(
+            new BanditLogEvent(
+                flagKey,
+                "??", // Bandit Key
```

Comment:
In the [java server sdk](https://github.com/Eppo-exp/java-server-sdk/blob/f2649304fb179fb6f933cf9e4453959be7b3dfbf/src/main/java/com/eppo/sdk/EppoClient.java#L489) the variation (result from `GetStringAssignment` above) is used here, but I think that's incorrect. The use case for this method is [_when a bandit action has not been assigned_](https://docs.geteppo.com/sdks/sdk-features/bandits/#logging-non-bandit-assignments), so the variation assigned here should not be a BanditKey.

Technically, we can look up the bandit key, and even get its model version by using the BanditVariation dto and the flag key provided here, but only because we _know, for now_ flags/bandits are 1:1, but that's not explicit in the data structures.

Reply:
Yeah if we proceed, thinking we just use the 1:1 flag to bandit mapping